sympy/parsing/c/c_parser.py

```diff
@@ -699,13 +668,11 @@ def transform_compound_stmt(self, node):
                 if the compound statement is empty
 
             """
-            try:
-                expr = []
-                children = node.get_children()
-                for child in children:
-                    expr.append(self.transform(child))
-            except StopIteration:
-                return None
+            expr = []
+            children = node.get_children()
+
+            for child in children:
+                expr.append(self.transform(child))
```

Comment:
Any good reason for removing the try block here? 
Exception handling is always good. Why remove it?

Reply:
The `get_children()` function returns an iterator, which is traversed by the `for` loop until the `StopIteration` exception is raised. As far as I can tell, the `try... except` is doing the same thing that the `for` loop is doing, but manually.